src/idl_parser.cpp

```diff
@@ -2616,6 +2625,10 @@ CheckedError Parser::ParseService() {
   service_def.file = file_being_parsed_;
   service_def.doc_comment = service_comment;
   service_def.defined_namespace = current_namespace_;
+  if (filename != nullptr && !opts.project_root.empty()) {
+    service_def.declaration_file =
```

Comment:
rather than storing yet another string in the `Definition`, would it be possible to compute these paths during serialization of the `.bfbs` such that we don't have to store anything we don't need?

Reply:
We can't just add the current filename argument to `Parser::Serialize` since there are included definitions in `Parser`. I think `std::set<std::string>` in `Parser` and `std::string&` in `Definition` would ameliorate the storage overhead. Do you think that would be worth the complexity?

Reply:
certainly having the string just allocated once seems like a win, yes.

Reply:
Okay, I added a generic string pool to `Parser`, but there are some similar opportunities for memory optimization that can be picked up in a future PR. Notably the following fields of `Parser`.

```
  std::map<uint64_t, std::string> included_files_;                                                                                                                                                                                                                                                                                                                          
  std::map<std::string, std::set<std::string>> files_included_per_file_;                                                                                                                                                                                                                                                                                                    
  std::vector<std::string> native_included_files_;      
```